pkg/controller/master-controller-manager/application-definition-synchronizer/controller.go

```diff
@@ -115,6 +116,16 @@ func (r *reconciler) reconcile(ctx context.Context, log *zap.SugaredLogger, requ
 	}
 
 	err := r.syncAllSeeds(log, applicationDef, func(seedClient ctrlruntimeclient.Client, appDef *appskubermaticv1.ApplicationDefinition) error {
+		seedDef := &appskubermaticv1.ApplicationDefinition{}
+		if err := seedClient.Get(ctx, ctrlruntimeclient.ObjectKeyFromObject(appDef), seedDef); err != nil && !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to fetch ClusterTemplate on seed cluster: %w", err)
```

Comment:
NIT: a few error messages have ClusterTemplate instead of the proper name

Reply:
Not a nit, but a real issue. Fixed now.